src/Tools/babylon.assetsManager.ts

```diff
@@ -318,12 +372,47 @@ module BABYLON {
         public runTask(scene: Scene, onSuccess: () => void, onError: (message?: string, exception?: any) => void) {
             scene._loadFile(this.url, (data) => {
                 this.text = data as string;
-                onSuccess();
+                if (this.isLoadAtlasImage)
+                    readJSONData(this.text);
+                else
+                    onSuccess();
             }, undefined, false, false, (request, exception) => {
                 if (request) {
                     onError(request.status + " " + request.statusText, exception);
                 }
             });
+
+            let readJSONData = (data: string) => {

```

Comment:
My point was that we should have no link st all to atlas data in this file. The load should be done by an atlas loader class here.
We cannot have a dependency to an external class (the atlas element)

We should have all code related to atlas in the Gui project:)

All this code should be in Gui and should rely on the TextFileAssetTask

Something like theanager you had before (plus a way to load the data without using the task ad not everyone wants to use the assets manager)

Something like (inside GUI):
- AtlasSourceDataHelper.Load(url)
- AtlasSourceDataHelper.PrepareAssetsManager

The second one will create a TextFileAssetTask and will use it to get the data. Then it will use the same code to get everything ready and returns a list of IAtlasSourceData

Does it make sense?

Reply:
Thanks for your review, i'm so stupid :( .
I will think carefully, slowly and try again.

Reply:
Man! Don't blame yourself
You are doing great. Adding code to a framework takes time and this is a perfectly natural process.
You are about to have a wonderful PR
Take you time, there is no rush